include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -127,10 +127,12 @@ class __histo_kernel_local_atomics;
 template <typename... _Name>
 class __histo_kernel_private_glocal_atomics;
 
-template <typename _HistAccessor, typename _OffsetT, typename _Size>
+template <typename _HistAccessor, typename _OffsetT, typename _Size,
+          typename _FenceSpace = decltype(__dpl_sycl::__fence_space_local)>
```

Comment:
Though... should we decay the type here?

Reply:
`__dpl_sycl::__fence_space_local` is either a `constexpr inline`  `sycl::access::fence_space` or `struct __fence_space_dummy{};

Does it change anything to decay `decltype(__dpl_sycl::__fence_space_local)`?  I would think it doesn't.  I suppose it doesn't hurt, but I'm not seeing the real danger of not decaying right now.

Reply:
I should say it will remove `const`, but I'm not sure why we want to do that here.

Reply:
One way to clear all this up would be to define a `__dpl__sycl::__fence_space_t` in the main PR and then using it in places where we are otherwise using `decltype(__dpl_sycl::__fence_space_local)`.  (here and in one place in the main PR I think)

Reply:
I added the definition of `__dpl__sycl::__fence_space_t` and associated changes.  I like it better, and it removes this question about decay.

@akukanov @dmitriy-sobolev what do you think?